src/extension.ts

```diff
@@ -1732,8 +1744,40 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
 }
 
 async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        promptKindName([kuberesources.allKinds.namespace], undefined,
+            {
+                prompt: 'What namespace do you want to use?',
+                placeHolder: 'Enter the namespace to switch to or press enter to select from available list'
+            },
+            (resource) => {
+                if (resource) {
+                    const currentNS = kubectlUtils.currentNamespace(kubectl);
+                    currentNS.then((ns) => {
```

Comment:
we try not to use `.then` or `.catch` in the code base, and prefer to use the more idiomatic `async/await` keywords.